packages/appstore/src/components/modals/wallet-modal/wallet-modal.scss

```diff
@@ -8,91 +8,158 @@
-    -ms-overflow-style: none;
-    /* Firefox */
-    scrollbar-width: none;
-    overflow-x: hidden;
-    overflow-y: scroll;
+.header-background {
+    width: 100%;
+    display: flex;
+    justify-content: center;
+    position: relative;
+    overflow: hidden;
+    background-color: #fbdddd;
 
-    &::-webkit-scrollbar {
-        display: none;
+    @include mobile {
+        position: fixed;
+        z-index: 3;
     }
+}
 
-    &::-webkit-scrollbar-thumb {
-        display: none;
-    }
+// styles for mobile and desktop modal header
+.modal-header {
+    max-width: 128rem;
+    width: 100%;
+    display: flex;
+    position: relative;
+    height: 16rem;
+    padding: 2.4rem 4rem 7.2rem;
+
+    @include mobile {
+        height: 12.2rem;
+        padding: 1.6rem 1.6rem 5.6rem;
+        transition: height 0.2s ease;
 
-    .dc-mobile-dialog {
-        &__header {
-            display: none;
+        .title-visibility {
+            height: 2rem;
         }
 
-        &__content {
-            margin: 0;
-            padding: 0;
+        .title-visibility,
+        .icon-visibility {
+            visibility: visible;
+            opacity: 1;
+            transition: visibility 0s, opacity 0.2s, height 0.2s ease;
+        }
+    }
 
-            .dc-mobile-dialog__wallet-mobile-dialog__wrapper {
-                .modal-header {
-                    height: 12.2rem;
-                    transition: height 0.2s ease;
-
-                    .title-visibility {
-                        height: 2rem;
-                    }
-
-                    .title-visibility,
-                    .icon-visibility {
-                        visibility: visible;
-                        opacity: 1;
-                        transition: visibility 0s, opacity 0.2s, height 0.2s ease;
-                    }
-                }
+    //TODO: check do we need this after bg change to radial-gradient
+    &__title-wrapper {
+        position: relative;
+    }
+
+    &__title {
+        display: flex;
+        align-items: center;
+
+        &-wallet {
+            padding-right: 0.8rem;
+            color: var(--text-general);
+
+            &-demo {
+                color: var(--demo-text-color-1);
             }
+        }
 
-            .scrolled-content {
-                .modal-header {
-                    height: 8.2rem;
+        &-balance {
+            color: var(--text-prominent);
 
-                    .title-visibility,
-                    .icon-visibility {
-                        visibility: hidden;
-                        opacity: 0;
-                        height: 0;
-                    }
-                }
+            &-demo {
+                color: var(--demo-text-color-2);
             }
         }
     }
+
+    &--hidden-title {
+        height: 8.2rem;
+        align-items: center;
+        justify-content: space-between;
+
+        .title-visibility,
+        .icon-visibility {
+            visibility: hidden;
+            opacity: 0;
```

Comment:
Why use visibility and opacity at the same time?

Reply:
@hamid-deriv removed the opacity property